tensorboard/data/server/gcs/auth.rs

```diff
@@ -235,6 +242,42 @@ impl Credentials {
     }
 }
 
+/// Partial structure of the `GOOGLE_APPLICATION_CREDENTIALS` file.
+#[derive(Deserialize)]
+struct ApplicationCreds {
+    r#type: Option<String>,
```

Comment:
The JSON field is called `type`, but `type` is a keyword in Rust, so we
use `r#type` as a *raw identifier*:
<https://doc.rust-lang.org/edition-guide/rust-2018/module-system/raw-identifiers.html>

An alternative would be to call the field something else and ask serde
to rename while deserializing:

```rust
#[derive(Deserialize)]
struct ApplicationCreds {
    #[serde(rename = "type")]
    creds_type: Option<String>,
    ...,
}
```

I don’t have a strong preference.